src/main/java/stream/flarebot/flarebot/Events.java

```diff
@@ -404,4 +412,23 @@ private void handleSpamDetection(GuildMessageReceivedEvent event, GuildWrapper g
             spamMap.put(event.getGuild().getId(), 1);
         }
     }
+
+    @Override
+    public void onGuildMessageDelete(GuildMessageDeleteEvent event) {
+        long messageID = event.getMessageIdLong();
+        if (removedByMe.contains(messageID)) {
+            removedByMe.remove(messageID);
+            return;
+        }
+        long discordEpoch = (messageID >> 22) + 1420070400000L;
+        long difference = System.currentTimeMillis() - discordEpoch;
+        this.durations.add(difference);
+        long dur = 0L;
+        int i = 0;
+        for (Long l : durations) {
+            dur += l;
+            i++;
+        }
+        avgDuration = dur / i;
```

Comment:
This is only used on shutdown, so why calculate it every single time a message is removed? Calculate it on shutdown

Reply:
+1